tracing-subscriber/src/filter/subscriber_filters/mod.rs

```diff
@@ -639,6 +639,22 @@ where
         }
     }
 
+    fn event_enabled(&self, event: &Event<'_>, cx: Context<'_, C>) -> bool {
+        let cx = cx.with_filter(self.id());
+        let enabled = FILTERING
+            .with(|filtering| filtering.and(self.id(), || self.filter.event_enabled(event, &cx)));
+
+        if enabled {
+            // If the filter enabled this event, ask the wrapped subscriber if
+            // _it_ wants it --- it might have a global filter.
+            self.subscriber.event_enabled(event, cx)
+        } else {
+            // Otherwise, return `true`. See the comment in `enabled` for why this
+            // is necessary.
+            true
+        }
```

Comment:
it occurs to me that this *could* be 
```suggestion
        // If the filter enabled this event, ask the wrapped subscriber if
        // _it_ wants it --- it might have a global filter.
        enabled && self.subscriber.event_enabled(event, cx)
```
(and we could make the same change in `enabled`)

but, this has the advantage that we can write comments in the `if` and the `else` branch, so i don't think we need to change it --- up to you!

Reply:
Using `&&`/`||` and side effectual conditions always feels iffy, so I have a mild preference for the long form here, especially since the arms both have comments "telling a story" about the branch. Actually using `&&`/`||` for control flow "just because" they line up with the desired boolean result feels too much like [a twitter shitpost](https://twitter.com/CAD97_/status/1547725877697339394?s=20&t=uEkIp2fPjFCKZH7qmV3ouA).

Either way, I think `event_enabled` *absolutely* should mirror the same structure as in `enabled`. And with `enabled`'s full comment (included by reference here), keeping the explicit `if` has more justification.

(With a time machine, though, I'd remove global filtering from `Subscribe` entirely, and implement global filtering by applying a `Filter` on the registry `Collect`or.)

Reply:
> Using `&&`/`||` and side effectual conditions always feels iffy, so I have a mild preference for the long form here, especially since the arms both have comments "telling a story" about the branch.

Yeah, I wasn't sure about this suggestion either --- I think using `&&` makes more sense in `and`, but I agree that this might be clearer; it wasn't a strong preference.

> (With a time machine, though, I'd remove global filtering from `Subscribe` entirely, and implement global filtering by applying a `Filter` on the registry `Collect`or.)

Yeah, I would have too. I wonder if that's something we should consider changing in a breaking release; it would certainly remove a lot of edge cases that we have to handle currently.